src/main/kotlin/apply/domain/mission/SubmissionMethod.kt

```diff
@@ -0,0 +1,6 @@
+package apply.domain.mission
+
+enum class SubmissionMethod(val label: String) {
```

Comment:
The choice of which method to use for representing the text associated with enum values in your `SubmissionMethod` enum in the `domain` package depends on the specific requirements and design considerations of your project. Each method has its own pros and cons, and the best choice will vary depending on the context. Here's an overview of each method with their pros and cons:

1. **The UI handles this:**
    - **Pros:**
        - Separates the text representation from the enum definition, which can be useful when translations or text changes are handled externally (e.g., by the UI).
        - Enum values are kept clean and concise.
    - **Cons:**
        - Might require external handling to get the text representation, which could make the code less self-contained.
2. **Public properties:**
    - **Pros:**
        - Easy access to the text representation from enum values.
        - Clear and self-contained definition of text representations.
    - **Cons:**
        - Enum values can become more verbose with additional properties.
3. **Function:**
    - **Pros:**
        - Provides a clear method for obtaining the text representation from enum values.
        - Keeps the enum values clean and concise.
    - **Cons:**
        - Slightly less convenient to access the text representation compared to public properties.
4. **Override `toString()`:**
    - **Pros:**
        - Leverages the built-in `toString()` method for getting the text representation, which can be convenient in some cases.
        - Concise enum value definitions.
    - **Cons:**
        - Overriding `toString()` for text representation might not be considered the best practice because `toString()` is typically used for debugging and not specifically for getting text representations.

Considering these pros and cons, here are some recommendations:

- If you expect the text representations to be handled outside the code (e.g., for localization), Method 1 or 2 could be a good choice as they separate the text from the enum values.
- If you want to keep the code self-contained and prefer clarity and easy access to the text representation, Method 2 or 3 might be suitable.
- Avoid using `toString()` for purposes other than debugging, as it might not be considered a good practice and could lead to confusion.

In summary, the choice between Method 2 and Method 3 may come down to personal or project-specific preferences. If you prioritize clear and self-contained code, Method 3 (using a function) is a good choice. However, if you prefer easy access to the text representations without the need to call a function, Method 2 (public properties) is a valid alternative.

Reply:
If you want a property name for managing Korean names in a way that doesn't depend on the View and is more generally applicable, I recommend using `label`. Here's why:

1. **`title`:** While "title" is commonly used for short, descriptive labels, it might not be as suitable for names in a non-UI context because names are typically more personal and substantial than titles.
2. **`text`:** "Text" is a general term, and it can be used for any textual content. However, it lacks the connotation of being a label or name, which could lead to ambiguity.
3. **`label`:** "Label" is a versatile term often used to describe a short, descriptive identifier for an item or a name. It's commonly used in non-UI contexts to represent identifiers or names for various objects, including enum values. "Label" is a good choice when you want to manage Korean names without being dependent on the View because it implies a concise and descriptive name.

So, in your case, if you're managing Korean names for enum values in a non-UI context, using a property named `label` is a suitable and clear choice.